base58.go

```diff
@@ -15,35 +15,59 @@ var bigRadix = big.NewInt(58)
 var bigZero = big.NewInt(0)
 
 // Decode decodes a modified base58 string to a byte slice.
-func Decode(b string) []byte {
-	answer := big.NewInt(0)
-	j := big.NewInt(1)
-
-	scratch := new(big.Int)
-	for i := len(b) - 1; i >= 0; i-- {
-		tmp := b58[b[i]]
-		if tmp == 255 {
+func Decode(input string) []byte {
+	if len(input) == 0 {
+		return []byte("")
+	}
+
+	// The max possible output size is when a base58 encoding consists of
+	// nothing but the alphabet character at index 0 which would result in the
+	// same number of bytes as the number of input chars.
+	output := make([]byte, len(input))
+
+	// Encode to base256 in reverse order to avoid extra calculations to
```

Comment:
base256?

Reply:
Nevermind, this is Decode. makes sense

Reply:
Yep, it is effectively base256.  There are 8 bits per byte.  2^8 = 256.